packages/bot-web-ui/src/stores/transactions-store.js

```diff
@@ -1,14 +1,24 @@
-import { action, observable } from 'mobx';
+import { action, observable, reaction } from 'mobx';
 import { formatDate } from '@deriv/shared';
 import { isEnded } from '@deriv/bot-skeleton';
 import { transaction_elements } from '../constants/transactions';
 
+const transaction_storage_key = 'transaction_cache';
+
 export default class TransactionsStore {
     constructor(root_store) {
         this.root_store = root_store;
+
+        reaction(
+            () => this.elements,
+            elements => sessionStorage.setItem(transaction_storage_key, JSON.stringify(elements))
+        );
     }
 
-    @observable elements = [];
+    @observable elements = Array.isArray(JSON.parse(sessionStorage.getItem(transaction_storage_key)))
+        ? JSON.parse(sessionStorage.getItem(transaction_storage_key))
+        : [];
```

Comment:
Same here, please try to reduce calls to `JSON.parse` and `sessionStorage.getItem` to once.